test/integration/main.sh

```diff
@@ -0,0 +1,141 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+SCRIPT_PATH=$(tmp=$(realpath "$0") && dirname "${tmp}")
+EXIT="0"
+IT_RUNNING="false"
+
+# install/uninstall/upgrade, utility stuff
+source ${SCRIPT_PATH}/common_functions.sh
+# cleanup functions
+source ${SCRIPT_PATH}/cleanup.sh
+
+# backup values.yaml
+cp charts/connaisseur/values.yaml charts/connaisseur/values.yaml.bak
+
+# cleanup on exit if not running in CI
+if [[ "${CI-}" != "true" ]]; then
+    trap_add "preserve_and_cleanup" SIGINT SIGTERM EXIT
+fi
+
+# update values.yaml with GHCR image
+if ghcr_case; then
+    ghcr_update "charts/connaisseur/values.yaml"
+fi
+
+IT_RUNNING="true"
+
+case ${1:-} in
+"regular")
+    # testing basic functionality of all validators
+    test_case "regular/install.yaml" "regular/cases.yaml" "make" "force"
+    ;;
+"notaryv1")
+    # testing notaryv1 validator
+    test_case "notaryv1/install.yaml" "notaryv1/cases.yaml" "make"
+    ;;
+"cosign")
+    # testing cosign validator
+    test_case "cosign/install.yaml" "cosign/cases.yaml" "make"
+    ;;
+"load")
+    # testing load
+    source ${SCRIPT_PATH}/load/test.sh
+    load_test
+    ;;
+"namespaced")
+    # testing namespace validation feature
+    source ${SCRIPT_PATH}/namespaced/test.sh
+    namespaced_validation_test
+    ;;
+"complexity")
+    # testing complex deployments and pods
+    # with multiple containers
+    source ${SCRIPT_PATH}/complexity/test.sh
+    complexity_test
+    ;;
+"deployment")
+    # testing different deployments with containers
+    # and init containers
+    test_case "deployment/install.yaml" "deployment/cases.yaml" "make"
+    ;;
+"pre-config")
+    # testing pre-configured values.yaml, without
+    # any changes, on deployments
+    install "make"
+    multi_test "pre-config/cases.yaml"
+    uninstall "make"
+    ;;
+"pre-config-and-workload")
+    # testing pre-configured values.yaml, without
+    # any changes, on deployments and on all workloads
+    # and different API versions
+    source ${SCRIPT_PATH}/workload/test.sh
+    pre_workload_test
+    ;;
+"cert")
+    # testing a custom TLS certificate 
+    # for connaisseur
+    source ${SCRIPT_PATH}/cert/test.sh
+    cert_test
+;;
+"redis-cert")
+    # testing a custom TLS certificate
+    # for redis
+    update_with_file "redis-cert/install.yaml"
+    install "make"
+    update_with_file "redis-cert/update.yaml"
+    upgrade "make"
```

Comment:
Didn't you (intend to) remove the ping on startup? If so, does this test actually do anything without explicitly verifying certs like `cert_test` does?

Reply:
changed the test so that the caching functionality is tested as well